### PR TITLE
add Minkowski Distance

### PR DESCRIPTION
**Pull request by jettcalleja**

### CODE REVIEW

File: test/others/minkowski-distance.spec.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  it('should return 2 with points (0, 1), (1, 1) in order 2.', function () {
+    expect(minkowskiDistance([0, 1], [1, 1], 2)).toBe(1);
+  });
+  it('should return 2 with points (0, 1, 4), (1, 1, 6) in order Positive Infinity.', function () {
+    expect(minkowskiDistance([0, 1, 4], [1, 1, 6], Number.POSITIVE_INFINITY)).toBe(2);
+  });
+  it('should return 0 with points (0, 1, 4), (1, 1, 6) in order Negative Infinity.', function () {
+    expect(minkowskiDistance([0, 1, 4], [1, 1, 6], Number.NEGATIVE_INFINITY)).toBe(0);
+  });
+  it('should return 8.372966759705923 with points (0, 3, 4, 5), (7, 6, 3, -1) in order 3.', function () {
```
Comment (mgechev) [inline test/others/minkowski-distance.spec.js]: Description doesn't match the content.

File: src/others/minkowski-distance.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+      return ret;
+    }
+
+    function minkowskiDistance (x, lx, y, ly, p) {
+      var d;
+      var i;
+
+      if (lx !== ly) {
+        throw '2 points must have same array length';
```
Comment (mgechev) [inline src/others/minkowski-distance.js]: Would you change it to "Both points must have same dimension"?

File: src/others/minkowski-distance.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      }
+
+      if (isNaN(p)) {
+        throw 'p must be a number';
```
Comment (mgechev) [inline src/others/minkowski-distance.js]: Would you change to it "The order "p" must be a number"?

File: src/others/minkowski-distance.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      }
+
+      if (p === Number.POSITIVE_INFINITY) {
+        return chebyshevDistance(x, y, lx, p, Math.max);
+      } else if (p === Number.NEGATIVE_INFINITY) {
+        return chebyshevDistance(x, y, lx, p, Math.min);
+      } else if (p < 1) {
+        throw 'p less than 1 will violate triangle inequality';
```
Comment (mgechev) [inline src/others/minkowski-distance.js]: Would you change it to: "Order less than 1 will violate the triangle inequality"?

**Comment by jettcalleja** [comment]
done with the requested changes. thanks for the review (Y)

**Pull request merged**